packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -212,11 +294,14 @@ class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTr
       parent: _currentController,
       curve: Curves.easeIn
     );
+    _currentAnimation.addListener(_onProgressChanged);
 
     // If we start out with a child, have the child appear fully visible instead
     // of animating in.
     if (widget.child != null)
       _currentController.value = 1.0;
+    else
```

Comment:
there's a comment above explaining the if statement but its explanation is now incomplete

Reply:
done